docs/import_hooks.md

```diff
@@ -0,0 +1,164 @@
+
+The package doesn't include any import hooks by default. You can add your own import hooks by adding a dictionary the the  `WAGTAIL_WORDPRESS_IMPORT_HOOKS` setting and defining a function thats will be called and passed the Page model and data to be imported.
+
+Following on with the example XML above, we can add an import hook to capture the `<item>` tags that are of type `attachment` and use that attachment data to create a header image for the blog post.
+
+### Import hook configuration
+
+Add a settings variable to your site settings file called `WAGTAIL_WORDPRESS_IMPORT_HOOKS` and add a dict with a key of `attachment` and values as follows:
+
+```python
+WORDPRESS_IMPORT_HOOKS_ITEMS_TO_CACHE = {
+    "attachment": {
+        "DATA_TAG": "_thumbnail_id",
+        "FUNCTION": "pages.import_hooks.header_image_importer",
+    }
+}
+```
+
+The `DATA_TAG` key is the `<wp:meta_key>_thumbnail_id</wp:meta_key>` value to be parsed. During the import process the key will be used to extract the value in the `<wp:meta_value>43120</wp:meta_value>` tag.
+
+The resulting dict item in the items_cache would be:
+
+
+
+```python
+{
+    ...
+    # each dict item is a nested dict that represents the XML item tags tree 
+    # of keys and values. It's not all shown here for brevity as 
+    # it can be a large amount of data.
+    ...
+    'wp:postmeta': [
+        ... # other wp:postmeta keys and values
+        {
+            'wp:meta_key': '_thumbnail_id', 
+            'wp:meta_value': 43120
+        }
+    ]
+}
+```
+
+The `FUNCTION` key is the dotted path to the function that will be called. The function should accept the Page model and the data to be imported as arguments. The function also needs to be passed an argument of `items_cache` whcih is the cache of all the items that have been imported with the `<wp:post_type>attachment</wp:post_type>`
+
+Sample function and page model for processing the header image:
+
+```python
+# pages/import_hooks.py
+
+"""The function called to process the header image"""
+
+def header_image_importer(page, data, items_cache):
+    # page.wp_post_meta[data] will exist if the Page model 
+    # inherits from WPImportedPageMixin
+    thumbnail_id = page.wp_post_meta[data]
+
+    # items_cache is a dict created automatically by the importer 
+    # when the import hook setting is included in your own settings
+    attachments = items_cache.get("attachment")
+
+    # the code below would be your own code to process the header image,
+    # you can perform any type of python/Wagtail logic here and make use 
+    # of thumbnail_id and attachments variables.
+    image_url = None
+
+    for attachment in attachments:
+        if (
+            attachment.get("wp:post_id")
+            and attachment.get("wp:post_id") == thumbnail_id
+        ):
+            image_url = attachment.get("guid")
+
+    if image_url:
+        image = get_or_save_image(image_url)
+        page.header_image = image
+        page.save()
```

Comment:
I will try this locally and thanks, thats made the JSONField query make sense to me now :)

Reply:
Thanks. On second thought, I'd change my line to assume a `pages` queryset was provided, rather than presupposing the use of ImportedPage here:

```
- pages = ImportedPage.objects.filter(**{lookup: thumbnail_id})
+ pages = pages.filter(**{lookup: thumbnail_id})
```

Reply:
In import authors the user example is now but it's not working here because the wp_post_meta is stored in a different way with a new function.

```python
def header_image(imported_pages, data, items_cache):
    """use this for the import authors version"""
    lookup = f"wp_post_meta__{data}"
    for attachment in items_cache:
        thumbnail_id = attachment.get("wp:post_id")
        pages = imported_pages.filter(**{lookup: thumbnail_id})
        if pages.exists():
            image = get_or_save_image(attachment.get("guid"))
            pages.update(header_image=image)
            print("Attaching header images to pages:", pages)
```